docs/quick_tutorial/authentication/tutorial/security.py

```diff
@@ -1,5 +1,21 @@
-USERS = {'editor': 'editor',
-         'viewer': 'viewer'}
+import os
+import hashlib
+import binascii
+
+SALT_LENGTH = 16
+
+def hash_password_with_salt(password, salt=None):
+    salt = salt or os.urandom(SALT_LENGTH)
+    dk = hashlib.pbkdf2_hmac('sha512', password.encode('utf-8'), salt, 100000)
+    return binascii.hexlify(dk + salt).decode('utf-8')
```

Comment:
No, this is not adjustable. hashlib.pbkdf2_hmac returns the bare hash, with neither salt nor iterations stored into the resulting hash. You cannot adjust the parameters with this or otherwise you cannot match the passwords later on!

I believe *if* we're wanting to show the proper way, then this example needs to install the passlib module and use the primitives from there instead.